patches/tModLoader/Terraria/ModLoader/Core/TmodFile.cs

```diff
@@ -436,4 +448,40 @@ private void Upgrade()
 		Open();
 		// Read contract fulfilled
 	}
+
+	internal bool VerifyHash()
+	{
+		IDisposable handle = null;
+		if (fileStream == null)
+			handle = Open();
```

Comment:
No need for the `if` check here. You need to use `using`. The guard statements should also run before trying to get a file handle.